features/functests/performance/performance.go

```diff
@@ -50,20 +63,145 @@ var _ = Describe("performance", func() {
+				"kernel.sched_rt_runtime_us":    "-1",
+				"vm.stat_interval":              "10",
+				"kernel.timer_migration":        "0",
+			}
+
+			nodes, err := clients.K8s.CoreV1().Nodes().List(metav1.ListOptions{
+				LabelSelector: perfWorkerNodesLabel,
 			})
 			Expect(err).ToNot(HaveOccurred())
+			Expect(len(nodes.Items)).Should(BeNumerically(">", 0), "cannot find nodes labeled as "+perfWorkerNodesLabel)
+
 			for _, node := range nodes.Items {
 				mcd, err := mcdForNode(&node)
 				Expect(err).ToNot(HaveOccurred())
 				mcdName := mcd.ObjectMeta.Name
-				bootLoaderEntries, err := exec.Command("oc", "rsh", "-n", "openshift-machine-config-operator", mcdName,
-					"grep", "-R", "initrd", "/rootfs/boot/loader/entries/").CombinedOutput()
+				for param, expected := range sysctlMap {
+					By(fmt.Sprintf("executing the command \"sysctl -n %s\" inside the pod %s", param, mcdName))
+					Eventually(func() string {
+						out, err := exec.Command("oc", "rsh", "-n", mcd.ObjectMeta.Namespace,
+							"-c", perfMachineConfigDaemonContainer, mcdName, "sysctl", "-n", param).CombinedOutput()
+						Expect(err).ToNot(HaveOccurred())
+						return strings.TrimSpace(string(out))
+					}, perfSysctlTimeout*time.Second, perfSysctlPollInterval*time.Second, ).Should(Equal(expected),
+						fmt.Sprintf("parameter %s value is not %s", param, expected))
+				}
+			}
+		})
+	})
+
+	var _ = Context("Essential RT kernel parameters", func() {
+		const perfSysctlKernelVersionParam = "kernel.version"
+		const perfSysctlKernelVersionValuePreemptEntry = "PREEMPT"
+		const perfSysctlKernelVersionValueRtEntry = "RT"
+		It(fmt.Sprintf("Should contain %s and %s entries",
+			perfSysctlKernelVersionValueRtEntry, perfSysctlKernelVersionValuePreemptEntry), func() {
+			nodes, err := clients.K8s.CoreV1().Nodes().List(metav1.ListOptions{
+				LabelSelector: perfWorkerNodesLabel,
+			})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(len(nodes.Items)).Should(BeNumerically(">", 0), "cannot find nodes labeled as "+perfWorkerNodesLabel)
+
+			for _, node := range nodes.Items {
+				mcd, err := mcdForNode(&node)
 				Expect(err).ToNot(HaveOccurred())
-				Expect(strings.Contains(string(bootLoaderEntries), "iso_initrd.img")).To(BeTrue())
+				mcdName := mcd.ObjectMeta.Name
+				By(fmt.Sprintf("executing the command \"sysctl -n %s\" inside the pod %s", perfSysctlKernelVersionParam, mcdName))
+				Eventually(func() bool {
+					out, err := exec.Command("oc", "rsh", "-n", mcd.ObjectMeta.Namespace,
+						"-c", perfMachineConfigDaemonContainer, mcdName, "sysctl", "-n", perfSysctlKernelVersionParam).CombinedOutput()
+					Expect(err).ToNot(HaveOccurred())
+					line := strings.TrimSpace(string(out))
+					return strings.Contains(line, perfSysctlKernelVersionValuePreemptEntry) &&
+						strings.Contains(line, perfSysctlKernelVersionValueRtEntry)
+				}, perfSysctlTimeout*time.Second, perfSysctlPollInterval*time.Second, ).Should(BeTrue(), "RT kernel is not installed")
```

Comment:
The last "," seems redundant or wrong .
In addition the async calls inside a loop using eventually might cause faulty behaviour here.